core/admin/mailu/internal/views/rspamd.py

```diff
@@ -25,3 +25,8 @@ def rspamd_dkim_key(domain_name):
                 }
             )
     return flask.jsonify({'data': {'selectors': selectors}})
+
+@internal.route("/rspamd/local_domains", methods=['GET'])
+def rspamd_local_domains():
+    domains = set(models.Domain.query.all() + models.Alternative.query.all())
+    return '\n'.join(map(str,domains))
```

Comment:
You should be able to use an iterator here:
```
return '\n'.join({str(domain) for domain in models.Domain.query.all() + models.Alternative.query.all()})
```

Wouldn't it be possible to create a /rspamd/local_domain/<domain> route which returns a bool. This would be better for installs with man domains.

Reply:
Hmm, architecturally I agree with you... but the rspamd module expects a "multimap" ... https://rspamd.com/doc/modules/multimap.html

I don't think it can be done using HTTP. We would have to fake a redis server or CDB adapter...

Alternatively the module could be rewritten in Lua... but my lua is rusty and again, I feel like this is premature optimization.

Extracting "just the name" out of sqlalchemy may be easier and get us most of the way there in terms of performance.

Reply:
The optimized query is at 84a722e